packages/key-readmodel-writer/src/keyConsumerServiceV1.ts

```diff
@@ -1,18 +1,69 @@
 import { KeyCollection } from "pagopa-interop-commons";
-import { AuthorizationEventEnvelopeV1 } from "pagopa-interop-models";
+import {
+  AuthorizationEventEnvelopeV1,
+  fromKeyV1,
+  toReadModelKey,
+  unsafeBrandId,
+} from "pagopa-interop-models";
 import { match } from "ts-pattern";
 
 export async function handleMessageV1(
   message: AuthorizationEventEnvelopeV1,
-  _keys: KeyCollection
+  keys: KeyCollection
 ): Promise<void> {
-  match(message)
+  await match(message)
+    .with({ type: "KeysAdded" }, async (message) => {
+      const keysToAdd = message.data.keys.map((keyEntry) => keyEntry.value);
+
+      for (const key of keysToAdd) {
+        if (key) {
+          await keys.updateOne(
+            {
+              "data.kid": key.kid,
+              "metadata.version": { $lt: message.version },
```

Comment:
Shouldn't we use `$lte` now after the problem that occurred a while ago?

Reply:
Thanks, I forgot to update this.